### PR TITLE
Add Asset class, ISearch interface. Update Search class to support ha…

### PR DESCRIPTION
**Pull request by shisova**
…ndling exceptions.

### CODE REVIEW

File: src/main/java/com/foundation/search/model/Search.java [hunk 1]
```diff
@@ -135,26 +144,42 @@ private boolean filterByContent(File file, String textToSearch){
...
         return addFileToSearchResult;
     }
 
+    /**
+     * getCreatedDate method allows to obtain the created date.
+     * @param file               Item (file or directory) that will be
+     *                           evaluated according to the search criteria.
+     * @return actualCreatedDate String value that contains the Created Date.
+     */
+	private String getCreatedDate(File file) throws IOException{
```
Comment (emelyllanos) [inline src/main/java/com/foundation/search/model/Search.java]: Please remove the space before "private"

File: src/main/java/com/foundation/search/model/Search.java [hunk 2]
```diff
@@ -167,26 +192,42 @@ private boolean filterByCreatedDate(File file, boolean isCreatedDateSelected,
...
-            } catch (IOException e) {
-                e.getMessage();
+            } catch (IOException e){
+                throw e;
             }
         }
         return addFileToSearchResult;
     }
 
+	/**
```
Comment (emelyllanos) [inline src/main/java/com/foundation/search/model/Search.java]: please remove the additional space before /**

File: src/main/java/com/foundation/search/model/Search.java [hunk 3]
```diff
@@ -199,26 +240,42 @@ private boolean filterByModifiedDate(File file, boolean isModifiedDateSelected,
...
```
Comment (emelyllanos) [inline src/main/java/com/foundation/search/model/Search.java]: please remove the additional space before /**

File: src/main/java/com/foundation/search/model/Asset.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+ * Copyright (c) 2018 Jalasoft.
+ * 2643 Av Melchor Perez de Olguin, Colquiri Sud, Cochabamba, Bolivia.
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information of
+ * Jalasoft, ("Confidential Information").  You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Jalasoft.
+ */
```
Comment (maryfranz) [inline src/main/java/com/foundation/search/model/Asset.java:11]: Remove the duplicate copyright

File: src/main/java/com/foundation/search/model/Search.java [hunk 5]
```diff
@@ -87,8 +90,8 @@
                 }
                 results.add(createSearchResult(file.getParent(),name,extension,
                     size,!file.canWrite(), file.isHidden(), file.isDirectory(),
-                    getCurrentOwner(file), criteria.getCreatedDate(),
-                    criteria.getModifiedDate(),criteria.getAccessedDate()));
+                    getCurrentOwner(file), getCreatedDate(file),
+                    getModifiedDate(file),getAccessedDate(file)));
```
Comment (maryfranz) [inline src/main/java/com/foundation/search/model/Search.java]: add space after comma

**Review by maryfranz — APPROVED** [review]

**Pull request merged**